azurerm/helpers/azure/app_service_schedule_backup.go

```diff
@@ -0,0 +1,207 @@
+								Type:         schema.TypeInt,
+								Optional:     true,
+								Default:      30,
+								ValidateFunc: validateRetentionPeriod,
+							},
+
+							"start_time": {
+								Type:             schema.TypeString,
+								Optional:         true,
+								DiffSuppressFunc: suppress.RFC3339Time,
+								ValidateFunc:     validate.RFC3339Time,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func validateFrequencyInterval(val interface{}, key string) (warns []string, errs []error) {
+	v := val.(int)
+
+	if v < 0 || v > 1000 {
+		errs = append(errs, fmt.Errorf("%q must be between 0 and 1000 inclusive, got: %d", key, v))
+	}
+	return
+}
+
+func validateRetentionPeriod(val interface{}, key string) (warns []string, errs []error) {
+	v := val.(int)
+
+	if v < 0 || v > 9999999 {
+		errs = append(errs, fmt.Errorf("%q must be between 0 and 9999999 inclusive, got: %d", key, v))
+	}
+	return
+}
+
+func ExpandAppServiceBackup(input []interface{}) *web.BackupRequest {
+	if len(input) == 0 {
+		return nil
+	}
+
+	vals := input[0].(map[string]interface{})
+
+	name := vals["name"].(string)
+	storageAccountUrl := vals["storage_account_url"].(string)
+	enabled := vals["enabled"].(bool)
+
+	request := &web.BackupRequest{
+		BackupRequestProperties: &web.BackupRequestProperties{
+			BackupName:        utils.String(name),
+			StorageAccountURL: utils.String(storageAccountUrl),
+			Enabled:           utils.Bool(enabled),
+		},
+	}
+
+	scheduleRaw := vals["schedule"].([]interface{})
+	if len(scheduleRaw) > 0 {
+		schedule := scheduleRaw[0].(map[string]interface{})
+		backupSchedule := web.BackupSchedule{}
+
+		if v, ok := schedule["frequency_interval"].(int); ok {
+			backupSchedule.FrequencyInterval = utils.Int32(int32(v))
+		}
+
+		if v, ok := schedule["frequency_unit"]; ok {
+			backupSchedule.FrequencyUnit = web.FrequencyUnit(v.(string))
+		}
+
+		if v, ok := schedule["keep_at_least_one_backup"]; ok {
+			backupSchedule.KeepAtLeastOneBackup = utils.Bool(v.(bool))
+		}
+
+		if v, ok := schedule["retention_period_in_days"].(int); ok {
+			backupSchedule.RetentionPeriodInDays = utils.Int32(int32(v))
+		}
+
+		if v, ok := schedule["start_time"].(string); ok {
+			dateTimeToStart, _ := time.Parse(time.RFC3339, v) //validated by schema
+			backupSchedule.StartTime = &date.Time{Time: dateTimeToStart}
+		}
+
+		request.BackupRequestProperties.BackupSchedule = &backupSchedule
+	}
+
+	return request
+}
+
+func FlattenAppServiceBackup(input *web.BackupRequestProperties) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	output := make(map[string]interface{})
+
+	if input.BackupName != nil {
+		output["name"] = *input.BackupName
+	}
+	if input.Enabled != nil {
+		output["enabled"] = *input.Enabled
+	}
+	if input.StorageAccountURL != nil {
+		output["storage_account_url"] = *input.StorageAccountURL
+	}
+
+	schedules := make([]interface{}, 0)
+	if input.BackupSchedule != nil {
+		v := *input.BackupSchedule
```

Comment:
might be a little friendly if we used schedule instead of v here:
```suggestion
		schedule := *input.BackupSchedule
```

Reply:
schedule's used immediately below - so this is intentionally different since the output objects use that (`schedule` & `schedules`)